src/sentry/spans/consumers/process/flusher.py

```diff
@@ -26,6 +27,74 @@
 logger = logging.getLogger(__name__)
 
 
+class MultiProducerManager:
+    """
+    Manages multiple Kafka producers for load balancing across brokers/topics.
+
+    Configure multiple producers in settings.py:
+
+    KAFKA_MULTI_PRODUCER_CONFIGS = {
+        "buffered-segments": [
+            {"cluster": "default", "topic": "buffered-segments-1"},
+            {"cluster": "secondary", "topic": "buffered-segments-2"}
+        ]
+    }
+    """
+
+    def __init__(self, topic: Topic):
+        self.topic = topic
+        self.producers: list[KafkaProducer] = []
+        self.topics: list[ArroyoTopic] = []
+        self.current_index = 0
+        self._setup_producers()
+
+    def _setup_producers(self):
+        """Setup producers based on configuration."""
+        # Get multi-producer config if available, otherwise use single producer
+        multi_config = getattr(settings, "KAFKA_MULTI_PRODUCER_CONFIGS", {}).get(self.topic.value)
+
+        if multi_config:
+            # Multiple producers configured
+            for config in multi_config:
+                cluster_name = config["cluster"]
+                topic_name = config.get(
+                    "topic", get_topic_definition(self.topic)["real_topic_name"]
+                )
+
+                producer_config = get_kafka_producer_cluster_options(cluster_name)
+                producer = KafkaProducer(build_kafka_configuration(default_config=producer_config))
+                topic = ArroyoTopic(topic_name)
+
+                self.producers.append(producer)
+                self.topics.append(topic)
+        else:
+            # Single producer (backward compatibility)
+            cluster_name = get_topic_definition(self.topic)["cluster"]
+            producer_config = get_kafka_producer_cluster_options(cluster_name)
+            producer = KafkaProducer(build_kafka_configuration(default_config=producer_config))
+            topic = ArroyoTopic(get_topic_definition(self.topic)["real_topic_name"])
+
+            self.producers.append(producer)
+            self.topics.append(topic)
+
+    def produce(self, payload: KafkaPayload, key: str | None = None):
+        """Produce message with load balancing."""
+        if len(self.producers) == 1:
+            # Single producer - no load balancing needed
+            return self.producers[0].produce(self.topics[0], payload)
+
+        # Round-robin load balancing
+        producer_index = self.current_index % len(self.producers)
+        self.current_index += 1
```

Comment:
Isn't the routing supposed to be based on the key?

Reply:
process-segments doesn't need it. we can randomly shuffle segments into the `buffered-segments` topic and redistribute the load. but this is also why this abstraction still lives in the span buffer, because it's not yet generic enough for other usecases.

if we didn't want to reshuffle and semantically partition the entire pipeline e2e by trace ID, then we would not need this routing in the buffer's producer.